### PR TITLE
Fixes #1943 - add polyline option for delineation by line

### LINKED ISSUE #1943: Change line delineation to polyline
- [x] Instead of 2 points that comprise a line, allow users to draw a polyline
- [x] Incorporate new endpoint: https://code.usgs.gov/StreamStats/web-services-and-apis/ss-pourpoint/-/issues/21
- [x] Edit text/instructions as needed

### PR DESCRIPTION
**Pull request by harper-wavra**
- option to delineate by polyline instead of a single line

If wanting to test before https://code.usgs.gov/StreamStats/web-services-and-apis/ss-pourpoint/-/merge_requests/16 is merged and deployed. You will need to run Pourpoint branch HW-21 locally and update the PourpointServices to 'http://127.0.0.1:8000' in the appconfig.

### CODE REVIEW

File: src/Views/sidebarview.html [hunk 1]
```diff
@@ -70,7 +70,7 @@ <h5>Supported search strings:</h5>
...
                             <p ng-show="vm.studyAreaService.zoomLevel15 && vm.studyAreaService.doDelineateFlag && vm.studyAreaService.delineateByPoint"><strong>Step 3:</strong> Use your mouse or finger to click or tap a blue stream cell on the map</p>
-                            <p ng-show="vm.studyAreaService.zoomLevel15 && vm.studyAreaService.doDelineateFlag && vm.studyAreaService.delineateByLine"><strong>Step 3:</strong> Use your mouse or finger to click or tap two points on the map to create a line. Blue stream cells that intersect the line will be delineated.</p>
+                            <p ng-show="vm.studyAreaService.zoomLevel15 && vm.studyAreaService.doDelineateFlag && vm.studyAreaService.delineateByLine"><strong>Step 3:</strong> Use your mouse or finger to click or tap points on the map to create a polyline. Double click to end the polyline. Blue stream cells that intersect the polyline will be delineated.</p>
```
Comment (amedenblik) [inline src/Views/sidebarview.html]: Is there a way to change the instructions in the map interface as well? It says "Click last point to finish line." which may be confusing.
image

File: src/Controllers/MapController.ts [hunk 2]
```diff
@@ -1048,7 +1045,7 @@ module StreamStats.Controllers {
                                     }
                                 }
                             });
-
+                            console.log(valid)
```
Comment (amedenblik) [inline src/Controllers/MapController.ts]:
```suggestion

```

File: src/Controllers/MapController.ts [hunk 3]
```diff
@@ -1027,7 +1022,9 @@ module StreamStats.Controllers {
                         map.invalidateSize();
                         var valid = true;
                         this.studyArea.lineIntersection(line).then((points) => {
+                            console.log(points)
                             points.forEach(point => {
+                                console.log(point)
```
Comment (amedenblik) [inline src/Controllers/MapController.ts]:
```suggestion

```

File: src/Controllers/MapController.ts [hunk 4]
```diff
@@ -1027,7 +1022,9 @@ module StreamStats.Controllers {
                         map.invalidateSize();
                         var valid = true;
                         this.studyArea.lineIntersection(line).then((points) => {
+                            console.log(points)
```
Comment (amedenblik) [inline src/Controllers/MapController.ts]:
```suggestion

```

**Review by amedenblik — CHANGES_REQUESTED** [review]
Works well!

**Comment by amedenblik** [comment]
We should redeploy SS-Pourpoint after this is merged, to include https://github.com/USGS-WiM/StreamStats/pull/1952

File: src/Views/sidebarview.html (hunk 1 above)
Comment (amedenblik) [reply src/Views/sidebarview.html]: We decided the default instructions are fine but are going to change "polyline" to "line" in the sidebar instructions.

**Review by amedenblik — APPROVED** [review]

**Comment by harper-wavra** [comment]
Deployed to https://dev.streamstats.usgs.gov/sc-runoff/

**Pull request merged**